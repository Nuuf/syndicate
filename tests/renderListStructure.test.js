/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('renders a structure', () => {
  const root = syndicate.createRoot();
  syndicate.add(root, syndicate.create({}, 'A'));
  syndicate.add(root, syndicate.create({}, 'B'));
  syndicate.add(root, syndicate.create({}, 'C'));
  syndicate.add(root, syndicate.create({}, 'D'));
  syndicate.add(root, syndicate.create({}, 'E'));
  syndicate.add(root, syndicate.create({}, 'F'));
  syndicate.remove(root, syndicate.find(root, 'F'));
  syndicate.remove(root, syndicate.find(root, 'A'));
  syndicate.remove(root, syndicate.find(root, 'C'));
  expect(syndicate.renderListStructure(root, true)).toMatchSnapshot();
});
