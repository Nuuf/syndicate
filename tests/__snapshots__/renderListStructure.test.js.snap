// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a structure 1`] = `
"id > root : index > 0 : level > 0
	id > B : index > 2 : level > 1
	id > D : index > 4 : level > 1
	id > E : index > 5 : level > 1
null > index 1
null > index 3
null > index 6
"
`;
